testrail/client.py

```diff
@@ -256,18 +256,22 @@ def _test_by_id(self, test_id, run):
             lambda t: t.raw_data()['case_id'] == test_id, self.tests(run))
 
     # Result Methods
-    def results(self, test_id):
-        return map(Result, self.api.results(test_id))
+    @methdispatch
+    def results(self):
+        raise TestRailError("Must request results by Run or Test")
+
+    @results.register(Run)
+    def _results_for_run(self, run):
+        return ResultContainer(list(map(Result, self.api.results_by_run(run.id))))
+
+    @results.register(Test)
+    def _results_for_test(self, test):
+        return ResultContainer(list(map(Result, self.api.results_by_test(test.id))))
 
     @methdispatch
     def result(self):
         return Result()
 
-    @result.register(int)
-    @singleresult
-    def _result_by_id(self, result_id):
-        return filter(lambda r: r.id == result_id, self.results())
```

Comment:
Note that I'm not sure this ever worked, as the original `results` method always required a test id. Additionally, results can only be queried if you supply a run, test or run/case combination. They can't be queried in bulk.